src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java

```diff
@@ -262,4 +269,18 @@ CompletableFuture<Result<CheckInContext>> findFulfillableRequest(CheckInContext
     return requestQueueService.findRequestFulfillableByItem(context.getItem(), context.getRequestQueue())
       .thenApply(r -> r.map(context::withHighestPriorityFulfillableRequest));
   }
+
+
+  CompletableFuture<Result<Request>> getDepartments(CheckInContext context) {
+    Request firstRequest = context.getHighestPriorityFulfillableRequest();
+    if (firstRequest == null || firstRequest.getRequester() == null) {
+      return completedFuture(succeeded(null));
+    }
+    List<String> departmentIds = firstRequest.getRequester().getDepartments()
+      .stream()
+      .map(String.class::cast)
+      .collect(Collectors.toList());
+    return CompletableFuture.completedFuture(Result.succeeded(firstRequest.withDepartments((departmentRepository.getDepartmentByIds(departmentIds)))));
```

Comment:
`departmentRepository.getDepartmentByIds` is an async operation which should return a CompletableFuture

Reply:
I changed the above method and now used the CqlIndexValuesFinder#findByIds method to fetch Departments